src/dashboard/src/components/accounts/views.py

```diff
@@ -51,15 +52,21 @@ def add(request):
             api_key = ApiKey.objects.create(user=newuser)
             api_key.key = api_key.generate_key()
             api_key.save()
+            user_profile = UserProfile.objects.get(user=newuser)
+            userprofileform = UserProfileForm(request.POST, instance=user_profile)
+            userprofileform.save()
 
             messages.info(request, _("Saved."))
             return redirect("components.accounts.views.list")
     else:
         # Clearing out values that are getting inherited from currently logged in user
         data = {"email": ""}
         form = UserCreationForm(initial=data)
+        userprofileform = UserProfileForm()
 
-    return render(request, "accounts/add.html", {"form": form})
+    return render(
+        request, "accounts/add.html", {"form": form, "userprofileform": userprofileform}
```

Comment:
If the `UserCreationForm` has a submission error (for example the password being too short), accessing `userprofileform` here results in a `referenced before assignment` exception:

```
UnboundLocalError at /administration/accounts/add/

local variable 'userprofileform' referenced before assignment
```